src/ripple/basics/SlabAllocator.h

```diff
@@ -100,91 +138,109 @@ class SlabAllocator
         SlabBlock(SlabBlock&& other) = delete;
         SlabBlock&
         operator=(SlabBlock&& other) = delete;
+    };
 
-        /** Determines whether the given pointer belongs to this allocator */
-        bool
-        own(std::uint8_t const* p) const noexcept
-        {
-            return (p >= p_) && (p < p_ + size_);
-        }
+    // The first slab, which is always present.
+    SlabBlock slab_;
 
-        std::uint8_t*
-        allocate() noexcept
-        {
-            std::uint8_t* ret;
+    // A linked list of slabs
+    std::atomic<SlabBlock*> slabs_ = nullptr;
 
-            {
-                std::lock_guard l(m_);
+    // A pointer to the slab we're attempting to allocate from
+    std::atomic<SlabBlock*> active_ = nullptr;
 
-                ret = l_;
+    // The size of an item, including the extra bytes requested and
+    // any padding needed for alignment purposes between items:
+    std::size_t const itemSize_;
 
-                if (ret)
-                {
-                    // Use memcpy to avoid unaligned UB
-                    // (will optimize to equivalent code)
-                    std::memcpy(&l_, ret, sizeof(std::uint8_t*));
-                }
-            }
+    // The maximum number of items that this slab can allocate
+    std::size_t const itemCount_;
 
-            return ret;
+    // The size of each individual slab:
+    std::size_t const slabSize_;
+
+    std::uint8_t*
+    allocate(SlabBlock* slab) noexcept
+    {
+        std::lock_guard l(slab->m_);
+
+        std::uint8_t* ret = slab->list_;
+
+        if (ret)
+        {
+            assert(
```

Comment:
are asserts checked at runtime in non-debug builds? I'm wondering if there are situations where this assert is not checked.